backends/cuda-shared/kernels/cuda-shared-basis-nontensor.cu

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and other CEED contributors.
+// All Rights Reserved. See the top-level LICENSE and NOTICE files for details.
+//
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This file is part of CEED:  http://github.com/ceed
+
+#include <ceed.h>
+#include <cuda.h>
+
+const int               MAX_SIZE = 52, MAX_DIM = 3;
+__constant__ CeedScalar c_B[MAX_SIZE * MAX_SIZE * MAX_DIM];
```

Comment:
This limit means `P * Q * dim` cannot exceed `52 * 52 * 3`... which is rather small, but there's only so much shared memory to go around

Reply:
We could fallback to ref in this case and warn that */gen is not going to work?

Reply:
Does the 52 limit include components? Ie for a quadratic line element, are we limited to `floor(52/3)` components?

Reply:
excluding components - we don't duplicate rows in the interp and grad matrices for components

Reply:
That number is device specific (no surprise there). I'm going to make an update to /cuda/shared to swap over to how /hip/shared handles this so we get a error instead of this silent memory access problem

Reply:
Ok, my most recent approach seems to help a lot